src/index.ts

```diff
@@ -470,14 +470,24 @@ export const DEFAULTS: RegisterOptions = {
 export class TSError extends BaseError {
   name = 'TSError';
   diagnosticText!: string;
+  diagnostics!: ReadonlyArray<_ts.Diagnostic>;
 
-  constructor(diagnosticText: string, public diagnosticCodes: number[]) {
+  constructor(
+    diagnosticText: string,
+    public diagnosticCodes: number[],
+    diagnostics: ReadonlyArray<_ts.Diagnostic> = []
```

Comment:
Is this argument optional to avoid a breaking change to our API surface, since `TSError` is part of our API?

Reply:
> Is this argument optional to avoid a breaking change to our API surface, since `TSError` is part of our API?

Yes, that's exactly why it's optional — I didn't want to make this a breaking change.